### PR TITLE
Handle new line characters as final characters

### PR DESCRIPTION
**Pull request by max-sixty**
As discussed in https://github.com/mitsuhiko/insta/issues/39

### CODE REVIEW

File: src/runtime.rs [hunk 1]
```diff
@@ -202,20 +198,12 @@ fn print_changeset(changeset: &Changeset, expr: Option<&str>) {
...
                 }
             }
         }
     }
-    println!(
-        "──────┴{:─^1$}",
-        "",
-        width.saturating_sub(7),
-    );
+    println!("──────┴{:─^1$}", "", width.saturating_sub(7),);
```
Comment (max-sixty) [inline src/runtime.rs]: these are auto changes from cargo fmt. I hadn't realized they were added

**Comment by max-sixty** [comment]
@mitsuhiko do you have any guidance on the best way forward here? 

@grovesNL made a reasonable suggestion at clearing this up, but not sure whether you have a clearer idea. 
I think ideally we want to do a comparison on the strings, rather than strings-transformed-by-`.lines.collect()` but not sure whether that's realistic given the existing code

**Comment by mitsuhiko** [comment]
I think it’s fine to change the behavior but we should add tests for all cases and make sure to document the behavior.

**Comment by max-sixty** [comment]
OK - I've simplified the changes down and added tests - let me know any feedback

**Comment by mitsuhiko** [comment]
This looks good! Will merge this.

**Pull request merged**